packages/artifact/src/upload-artifact-http-client.ts

```diff
@@ -0,0 +1,299 @@
+import {debug} from '@actions/core'
+import {BearerCredentialHandler} from '@actions/http-client/auth'
+import {HttpClientResponse, HttpClient} from '@actions/http-client/index'
+import {IHttpClientResponse} from '@actions/http-client/interfaces'
+import {
+  CreateArtifactResponse,
+  CreateArtifactParameters,
+  PatchArtifactSize,
+  PatchArtifactSizeSuccessResponse
+} from './contracts'
+import * as fs from 'fs'
+import {SearchResult} from './search'
+import {URL} from 'url'
+import {
+  parseEnvNumber,
+  getArtifactUrl,
+  isSuccessStatusCode,
+  isRetryableStatusCode,
+  getRequestOptions,
+  getContentRange
+} from './utils'
+
+const defaultChunkUploadConcurrency = 3
+const defaultFileUploadConcurrency = 2
+
+/**
+ * Step 1 of 3 when uploading an artifact. Creates a file container for the new artifact in the remote blob storage/file service
+ * @param {string} artifactName Name of the artifact being created
+ * @returns The response from the Artifact Service if the file container was succesfully created
+ */
+export async function createArtifactInFileContainer(
+  artifactName: string
+): Promise<CreateArtifactResponse> {
+  const token = process.env['ACTIONS_RUNTIME_TOKEN'] || ''
+  const bearerCredentialHandler = new BearerCredentialHandler(token)
+  const requestOptions = getRequestOptions()
+  requestOptions['Content-Type'] = 'application/json'
+
+  const client: HttpClient = new HttpClient('actions/artifact', [
+    bearerCredentialHandler
+  ])
+  const parameters: CreateArtifactParameters = {
+    Type: 'actions_storage',
+    Name: artifactName
+  }
+  const data: string = JSON.stringify(parameters, null, 2)
+  const rawResponse: HttpClientResponse = await client.post(
+    getArtifactUrl(),
+    data,
+    requestOptions
+  )
+
+  const body: string = await rawResponse.readBody()
+  const response: CreateArtifactResponse = JSON.parse(body)
+  console.log(response)
+
+  if (rawResponse.message.statusCode === 201 && response) {
+    return response
+  } else {
+    throw new Error(
+      'Non 201 status code when creating file container for new artifact'
```

Comment:
1. I don't think we should exclusively check `201`, since any 2xx response is a successful response status
2. The error should contain the failed response status code

Reply:
Our API is setup to return a 201. After a lot of testing, I've only seen a a 201 returned if there is ever anything between 200-299. 

That being said, the behavior of our APIs might change and we could return something like a 200 in the future so I've updated it to work with any success code. 

Also, in the event of a non-success code it just logs the entire raw response so there should be plenty of debug info